### PR TITLE
Neo4j 4.x syntax autopatcher, optionally disable with a decorator

### PR DESCRIPTION
**Pull request by ramonpetgrave64**
Continuing on #861 

We're adding a decorator you can use to disable the syntax upgrader for any functions you want to explicitly disable the syntax upgrader.

### CODE REVIEW

**Review by achantavy — COMMENTED** [review]
Hey, do we need this merged in? Can you share an invocation example and scenario of when you would want to use this?

**Comment by ramonpetgrave64** [comment]
> Hey, do we need this merged in? Can you share an invocation example and scenario of when you would want to use this?

Added here. https://github.com/lyft/cartography/pull/866/files#diff-f46377f43949b669f9290eab5ffaacd38270a3b873ab60df84850deeb9c5931fR18-R25

File: cartography/experimental_neo4j_4x_support.py [hunk 1]
```diff
@@ -78,6 +87,25 @@ def patch_libraries():
...
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        global UPGRADE_SYNTAX
+        original = UPGRADE_SYNTAX
+        UPGRADE_SYNTAX = False
+        try:
+            return_value = func(*args, **kwargs)
+            UPGRADE_SYNTAX = original
+            return return_value
+        # except Exception as e:
```
Comment (skiptomyliu) [inline cartography/experimental_neo4j_4x_support.py:101]: remove commented out code

File: cartography/experimental_neo4j_4x_support.py [hunk 2]
```diff
@@ -78,6 +87,25 @@ def patch_libraries():
...
+        except Exception:
+            UPGRADE_SYNTAX = original
+            # print(e)
```
Comment (skiptomyliu) [inline cartography/experimental_neo4j_4x_support.py:104]: remove commented out code

**Review by skiptomyliu — APPROVED** [review]

**Review by ruwaifaa — APPROVED** [review]

**Pull request merged**